routing/single_mwm_router.cpp

```diff
@@ -255,17 +230,28 @@ bool SingleMwmRouter::RedressRoute(MwmSet::MwmId const & mwmId, IVehicleModel co
 
 // static
 unique_ptr<SingleMwmRouter> SingleMwmRouter::CreateCarRouter(
-    Index const & index, traffic::TrafficCache const & trafficCache)
+    Index & index, TCountryFileFn const & countryFileFn, shared_ptr<NumMwmIds> numMwmIds,
+    traffic::TrafficCache const & trafficCache)
 {
   auto vehicleModelFactory = make_shared<CarModelFactory>();
   // @TODO Bicycle turn generation engine is used now. It's ok for the time being.
   // But later a special car turn generation engine should be implemented.
-  auto directionsEngine = make_unique<BicycleDirectionsEngine>(index);
-  auto estimator =
-      EdgeEstimator::CreateForCar(*vehicleModelFactory->GetVehicleModel(), trafficCache);
-  auto router =
-      make_unique<SingleMwmRouter>("astar-bidirectional-car", index, trafficCache,
-                                   move(vehicleModelFactory), estimator, move(directionsEngine));
+  auto directionsEngine = make_unique<BicycleDirectionsEngine>(index, numMwmIds);
+
+  double maxSpeed = 0.0;
+  numMwmIds->ForEachId([&](NumMwmId id) {
+    string const & country = numMwmIds->GetFile(id).GetName();
+    double const mwmMaxSpeed =
+        vehicleModelFactory->GetVehicleModelForCountry(country)->GetMaxSpeed();
+    maxSpeed = max(maxSpeed, mwmMaxSpeed);
+  });
+
+  auto trafficStash = make_shared<TrafficStash>(trafficCache);
+
+  auto estimator = EdgeEstimator::CreateForCar(trafficStash, maxSpeed);
```

Comment:
Up to you of course, but why not to just return nullptr here and let the caller to decide what to do?